cumulus/pallets/parachain-system/src/lib.rs

```diff
@@ -1722,7 +1723,10 @@ pub struct RelayChainState {
 /// This exposes the [`RelayChainState`] to other runtime modules.
 ///
 /// Enables parachains to read relay chain state via state proofs.
-pub trait RelaychainStateProvider {
+pub trait RelaychainStateProvider<BlockNumber, Hash> {
+	/// Returns the block hash of a relay chain block at a specific block number.
+	fn block_hash_at(at: BlockNumber) -> Hash;
```

Comment:
This trait is for providing the state and not pieces from this state.

Reply:
Should I create a new trait specifically for this? Or is there another trait where I could add this?

Reply:
The `RelayChainStateProof` contains some functions for reading the state. But you also don't want to include all the block hashes stored on the relay chain in the proof. The best would be that you provide the proof alongside the core claim to the runtime. Or somehow ensure that only this block hash is included in the proof.

Reply:
I don't really see how the proof would include all the block hashes. AFAIK, the Base-16 trie is used as a storage structure, so it wouldn't need all the other block hashes in the proof.

Reply:
The point is that you want to read the block hash in the runtime. However, the proof is produced before on the node side. You will need to know when producing the proof, which hash will be read.